api/v1alpha1/osbuild_types.go

```diff
@@ -90,7 +81,17 @@ type OSBuildCondition struct {
 	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty" description:"last time the condition transit from one status to another"`
 }
 
-type OSBuildConditionType string
+type ConditionType string
+
+// These are the valid OSBuild condition types
+const (
+	// Whether OSBuild build is done and ready
```

Comment:
The name of the const is general, so its documentation should also be general (so that we can use it in other controllers, e.g. OSBuildEnvConfig). Something like `The resource is ready`, `The resource failed`, `The resource is in progress`. 
We could also change the name and make it specific to OSBuild, but then we will need to duplicate the values